### PR TITLE
Expose all the modules

### PR DESCRIPTION
**Pull request by amosr**
It would be very useful to be able to implement other parsers, and exposing all the modules would make this easier/possible.

### CODE REVIEW

**Comment by bgamari** [comment]
If we are going to do this then let's move these modules to the `.Internal` namespaces. I really don't want to see these cluttering the documentation

File: attoparsec.cabal [hunk 1]
```diff
@@ -69,7 +69,7 @@ library
                    Data.Attoparsec.Text.Lazy
                    Data.Attoparsec.Types
                    Data.Attoparsec.Zepto
-  other-modules:   Data.Attoparsec.ByteString.Buffer
+                   Data.Attoparsec.ByteString.Buffer
```
Comment (mbj) [inline attoparsec.cabal]: You likely want to sort the merged export list.

**Comment by amosr** [comment]
Sounds good. It is a bit odd because at the moment there are "Data.Attoparsec.Internal.X" and "Data.Attoparsec.X.Internal". Shall I rename them to be consistent while I'm at it - and if so do you have a preference?

**Comment by amosr** [comment]
Will move these modules into internal and create a new PR in a few weeks

**Pull request closed without merging**